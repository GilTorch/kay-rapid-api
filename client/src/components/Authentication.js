import React from 'react';
import { Link } from 'react-router-dom';
import "../css/authentication.css";

class Authentication extends React.Component{
    constructor(props){
        super(props);
    }

    render(){
        return(
        <div className="card signin-card">
            <div className="signin-card__wrapper">
                <p className="signin-card__message">Konekte kounya pou ou ka genyen yon pi bon eksperyans.</p>
                <Link to="/authentication/sign-in-with-social-media">                                                    
                    <button className="auth-button success-button">KONEKTE</button>
                </Link>
            </div>
        </div>
        )
    }
}

export default Authentication;