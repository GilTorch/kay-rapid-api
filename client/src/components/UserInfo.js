import React from 'react';
import ImageLoader from 'react-image-file';

const UserInfo=({ firstName,lastName,profilePicture })=>{
   return( 
    <div className="profile-info">
        <ImageLoader file={profilePicture}/>
        <p>{firstName} {lastName}</p>
    </div>
   )
}

export default UserInfo;