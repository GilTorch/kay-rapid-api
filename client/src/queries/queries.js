import gql from "graphql-tag";

export const ALL_HOUSES_QUERY=gql`
  query {
    Houses{
      id 
      host{
        id 
        firstName
        lastName
        phone1 
      }
      numBedrooms
      numBaths 
      numDiningrooms 
      pricing{
        highestPrice 
        currency
      }
      leaseType 
      location{
        address
      }
      preview_image{
        url
      }
    }
  }
`

export const ACCOUNT_CREATION = gql`
mutation signup($email:String!,$password:String!,$firstName:String!,$lastName:String!,$phone1:String!,$profilePicture:String!){
  signup(email:$email,password:$password,firstName:$firstName,lastName:$lastName,phone1:$phone1,profilePicture:$profilePicture){
  token
  user{
    id
  }
  }
}
` 

export const REQUEST_PASSWORD_TOKEN=gql`
  mutation requestPWResetToken($email:String!){
    requestPWResetToken(email:$email){
      message
    }
  }
`

export const RESET_PASSWORD=gql`
mutation resetPassword($resetToken:String!,$password:String!,$passwordConfirm:String!){
  resetPassword(resetToken:$resetToken,password:$password,passwordConfirm:$passwordConfirm){
    token
    user{
      id
    }
  }
}
`

export const AUTH_WITHOUT_SOCIAL_MEDIA=gql`
mutation login($email:String!,$password:String!) {
  login(email:$email,password:$password) {
    token 
    user{
      id
      firstName
      lastName
      email
      profilePicture{
        url
      }
    }
  }
}
`


export const HOUSE_CREATION_2=gql`
  mutation createHouse($age:Int!,$maxGuests:Int!,$numBedrooms:Int!,$numBaths:Int!,$basePrice:Int!,$numLivingrooms:Int!,$numDiningrooms:Int!,$highestPrice:Int!,$income:Int!,$lease:Int!,$communeId:String, $currency:CURRENCY){
    createHouse(age:$age,maxGuests:$maxGuests,numBedrooms:$numBedrooms,numBaths:$numBaths,basePrice:$basePrice,numLivingrooms:$numLivingrooms,numDiningrooms:$numDiningrooms,highestPrice:$highestPrice,income:$income,lease:$lease,communeId:$communeId,currency:$currency){
      id
    }
  }
`

export const HOUSE_CREATION=gql`
  mutation createHouse(
    $area:Int,
    $age:Int!,
    $shortDescription:String,
    $description:String,
    $maxGuests:Int!,
    $numBedrooms:Int!,
    $numBaths:Int!,
    $basePrice:Int!,
    $numLivingrooms:Int!,
    $numDiningrooms:Int!,
    $highestPrice:Int!,
    $currency:CURRENCY,
    $income:Int!,
    $lat:Float,
    $lng:Float,
    $address:String,
    $communeId:String,
    $leaseType:HOUSE_LEASE_TYPE,
    $type:HOUSE_TYPES,
    $residency:House_RESIDENCY,
    $lease:Int!,
    $electricity:Boolean,
    $electricity_frequency:FREQUENCY,
    $water_pipe:Boolean,
    $water_tank:Boolean,
    $water_frequency:FREQUENCY,
    $elevator:Boolean,
    $petsAllowed:Boolean,
    $internet:Boolean,
    $kitchen:Boolean,
    $wirelessInternet:Boolean,
    $familyKidFriendly:Boolean,
    $freeParkingOnPremises:Boolean,
    $hotTub:Boolean,
    $pool:Boolean,
    $smokingAllowed:Boolean,
    $wheelchairAccessible:Boolean,
    $cableTv:Boolean,
    $suitableForEvents:Boolean,
    $dryer:Boolean,
    $washer:Boolean,
    $indoorFireHouse:Boolean,
    $tv:Boolean,
    $hangers:Boolean,
    $iron:Boolean,
    $hairDryer:Boolean,
    $doorman:Boolean,
    $paidParkingOffPremises:Boolean,
    $freeParkingOnStreet:Boolean,
    $gym:Boolean,
    $airConditioning:Boolean,
    $shampoo:Boolean,
    $essentials:Boolean,
    $laptopFriendlyWorkspace:Boolean,
    $privateEntrance:Boolean,
    $buzzerWirelessIntercom:Boolean,
    $bathtub:Boolean,
    $crib:Boolean
    $rooms:[RoomCreateWithoutHouseInput],
    $previewImage:String
    ){
    createHouse(
    area:$area,
    age:$age,
    shortDescription:$shortDescription,
    description:$description,
    maxGuests:$maxGuests,
    numBedrooms:$numBedrooms,
    numBaths:$numBaths,
    basePrice:$basePrice,
    numLivingrooms:$numLivingrooms,
    numDiningrooms:$numDiningrooms,
    highestPrice:$highestPrice,
    currency:$currency,
    income:$income,
    lat:$lat,
    lng:$lng,
    address:$address,
    communeId:$communeId,
    leaseType:$leaseType,
    type:$type,
    residency:$residency,
    lease:$lease,electricity:$electricity,
    electricity_frequency:$electricity_frequency,
    water_pipe:$water_pipe,
    water_tank:$water_tank,
    water_frequency:$water_frequency,
    elevator:$elevator,
    petsAllowed:$petsAllowed,
    internet:$internet,
    kitchen:$kitchen,
    wirelessInternet:$wirelessInternet,
    familyKidFriendly:$familyKidFriendly,
    freeParkingOnPremises:$freeParkingOnPremises,
    freeParkingOnStreet:$freeParkingOnStreet,
    hotTub:$hotTub,
    pool:$pool,
    smokingAllowed:$smokingAllowed,
    wheelchairAccessible:$wheelchairAccessible,
    cableTv:$cableTv,
    suitableForEvents:$suitableForEvents,
    dryer:$dryer,
    washer:$washer,
    indoorFireHouse:$indoorFireHouse,
    tv:$tv,
    hangers:$hangers,
    iron:$iron,
    hairDryer:$hairDryer,
    doorman:$doorman,
    paidParkingOffPremises:$paidParkingOffPremises,
    gym:$gym,
    airConditioning:$airConditioning,
    shampoo:$shampoo,
    essentials:$essentials,
    laptopFriendlyWorkspace:$laptopFriendlyWorkspace,
    privateEntrance:$privateEntrance,
    buzzerWirelessIntercom:$buzzerWirelessIntercom,
    bathtub:$bathtub,
    rooms:$rooms
    crib:$crib,
    previewImage:$previewImage){
      id,
      rooms {
      id,
      label,
      picture_previews{
      	url
      }
    }
      
    }
  }
`

export const FB_AUTH = gql`
mutation authenticateFBUser($facebookToken:String!) {
authenticateFBUser(facebookToken:$facebookToken){
  token
  user{
    id
    firstName
    lastName
    email
    profilePicture{
      url
    }
  }
}
}
`

export const WRITE_AUTH_INFO= gql`
mutation updateAuthInfo($userAuthInfo:Object!){
  updateAuthInfo(userAuthInfo:$userAuthInfo) @client{
    id
    token
    email
    firstName
    lastName
    # profilePicture
  }
}
`

export const READ_AUTH_INFO = gql`
  query{
    userAuthInfo @client{
      id
      token
      email
      firstName
      lastName
      profilePicture
    }
  }
`
