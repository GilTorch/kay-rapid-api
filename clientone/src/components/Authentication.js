import React from 'react';
import { Link } from 'react-router-dom';

class Authentication extends React.Component{
   

    render(){
        return(
        <div className="card signin-card">
            <div className="signin-card__wrapper">
                <p>Konekte kounya pou ou ka genyen yon pi bon eksperyans.</p>
                <Link to="/authentication/sign-in-with-social-media">                                                    
                    <button className="authentication__button signin-card__button">KONEKTE</button>
                </Link>
            </div>
        </div>
        )
    }
}

export default Authentication;