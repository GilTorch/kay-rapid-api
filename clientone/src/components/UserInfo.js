import React from 'react';


const UserInfo=({ firstName,lastName,profilePicture })=>{
   return( 
    <div className="profile-info" style={{marginTop:"100px"}}>
        <img src={profilePicture}/>
        <p>{firstName} {lastName}</p>
        <button className="authentication-button profile-info__logout-button">Log Out</button>
    </div>
   )
}

export default UserInfo;