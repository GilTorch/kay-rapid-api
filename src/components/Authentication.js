import React from 'react';
import SignIn from './SignIn';
import { BrowserRouter as Router,Route } from 'react-router-dom';
import SignInWithSocialMedia from './SignInWithSocialMedia';
import SignInWithoutSocialMedia from './SignInWithoutSocialMedia';
import AccountCreation from './AccountCreation';


class Authentication extends React.Component{

    constructor(props){
        super(props);
        this.closeCard=this.closeCard.bind(this);
        // $('.authentication__signin-with-socialmedia').hide();
        // $('.authentication__signin-without-socialmedia').hide();
        // $('.authenctication__account-creation').hide();
        this.state={
            userIsSignedIn:false,
        }
    }


    closeCard(){
        
    }

    render(){
       return(
        <div className="authentication">
        <Router>
        <React.Fragment>
          <Route exact={true} path="/authentication" component={SignIn}/>
          <Route exact={true} path="/authentication/sign-in-with-social-media" component={SignInWithSocialMedia}/>
          <Route exact={true} path="/authentication/sign-in-without-social-media" component={SignInWithoutSocialMedia}/>
          <Route exact={true} path="/authentication/account-creation" component={AccountCreation}/>
        </React.Fragment>
      </Router>
        </div>
        )
    }

}


export default Authentication;