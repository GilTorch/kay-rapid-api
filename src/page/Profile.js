import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Navigation from '../components/Navigation';
import HeaderBar from '../components/HeaderBar';
import Authentication from '../components/Authentication';




class Profile extends React.Component{
    

    render(){
        return(
            <div className="profile">
                <HeaderBar title="Profile"/>
                <Authentication/>
                <Navigation currentPage="profile"/>
            </div>
        )
    }
   
}  

export default Profile;