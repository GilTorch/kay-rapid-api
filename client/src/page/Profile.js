import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Navigation from '../components/Navigation';
import HeaderBar from '../components/HeaderBar';
import Authentication from '../components/Authentication';

import { graphql,Query } from 'react-apollo';
import { READ_AUTH_INFO } from '../queries/queries';
import UserInfo from '../components/UserInfo';

import { withRouter } from 'react-router-dom';

import "../css/profile.css";

import Loading from '../components/Loading';

import notify from '../utils/notify';
import { toast } from 'react-toastify';



// class Profile extends React.Component{
    
//     constructor(props){
//         super(props)
//     }

//     render(){
//             const {userAuthInfo}=this.props;
//             let componentToRender;
            
//             if(userAuthInfo){
//                 if(userAuthInfo.email)
//                 {
//                     notify('OU KONEKTE',"success")
//                     componentToRender=<UserInfo {...userAuthInfo} />
//                 }
//                 else{
//                     notify('OU PA KONEKTE',"error")
//                     componentToRender=<Authentication/>
//                 }
//             }

//             return(
//                 <div>
//                     <HeaderBar title="Pwofil Ou"/>
//                     {componentToRender}
//                     <Navigation currentPage="profile"/>
//                 </div>
//             )
//     }
// }  

// const ProfileWithQuery=graphql(
//     READ_AUTH_INFO,{
//         props:({data:{ userAuthInfo }})=>({
//             userAuthInfo
//         })
//     }
// )(Profile)


// export default withRouter(ProfileWithQuery) ;

// const UserInfoWrappedInHeader
{/* <HeaderBar title="Pwofil Ou"/>
            {componentToRender}
        <Navigation currentPage="profile"/>

        
let componentToRender;

if (userAuthInfo){
    if(userAuthInfo.email){
        toast.success("OU KONEKTE",{autoClose:1000})
        componentToRender=<UserInfo {...this.props.info}/>
    }else{
        toast.error("OU PA KONEKTE",{autoClose:1000})
        componentToRender=<Authentication/>
    }
} */}

// const { userAuthInfo }=this.props;

const Profile=()=>(            
    <Query query={READ_AUTH_INFO}>
    {({loading,error,data})=>{
        if (loading) return <Loading/>;
        if (error) {
            toast.error(error.message,{autoClose:2000});
        }
        if(data.userAuthInfo.email){
            toast.success("OU KONEKTE",{autoClose:800})
            return(
                <React.Fragment>
                    <HeaderBar title="Pwofil Ou"/>
                        <UserInfo {...data.userAuthInfo} />
                    <Navigation currentPage="profile"/>
                </React.Fragment>
            )
        }else{
            toast.warn("OU PA KONEKTE",{autoClose:1200})
            return(
                <React.Fragment>
                    <HeaderBar title="Pwofil Ou"/>
                        <Authentication/>
                    <Navigation currentPage="profile"/>
                </React.Fragment>
            )
        }
        
    }}
    </Query>
)

export default Profile;