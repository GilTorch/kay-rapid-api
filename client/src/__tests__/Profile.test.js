import React from 'react';
import Profile from '../page/Profile';
import Authentication from '../components/Authentication';
import UserInfo from '../components/UserInfo';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'
import { MockedProvider } from 'react-apollo/test-utils';
import { READ_AUTH_INFO } from '../queries/queries';
import { shallow,mount } from 'enzyme';

describe('<Profile/>',()=>{

    const wait = (time = 0) => new Promise(res => setTimeout(res, time));
    const executeMockProviderTestCase = (wrapperInstance) => {
        return wait(100).then(() => wrapperInstance.update());
    }

    const mocks = [
        {
          request: {
            query: READ_AUTH_INFO,
          },
          result: {
            data: {
                userAuthInfo:{email:null,firstName:null,lastName:null,profilePicture:null,token:null},
            },
          },
        },
      ];
    
    
    it('renders correctly', () => {
        const wrapper=mount(
        <MockedProvider mocks={mocks} addTypeName={false} >
            <Profile />
        </MockedProvider>
        )
    //   const tree = renderer
    //     .create(
    //         <MemoryRouter  initialEntries={['/profile']}>
    //             <MockedProvider mocks={mocks} addTypeName={false} >
    //                 <Profile />
    //             </MockedProvider>
    //         </MemoryRouter>
    //     )
    //     .toJSON();
      expect(wrapper).toBeDefined;
    });
    
    it('renders a <Authentication/> component when user is not logged in',()=>{
//         const wrapper = shallow(
//             <MemoryRouter  initialEntries={['/profile']}>            
//                 <MockedProvider mocks={mocks} addTypeName={false} >
//                     <Profile userAuthInfo={{email:null,firstName:null,lastName:null,profilePicture:null,token:null}}
//  />
//                 </MockedProvider>
//             </MemoryRouter>
//         )
//         return executeMockProviderTestCase(wrapper).then(() => {
//             expect(wrapper.find(".signin-card").length).toEqual(1);
//           });
        expect(1).toEqual(1)
    })
    
    
})
// userAuthInfo={{__typename:"UserAuthInfo",email:null,firstName:null,lastName:null,profilePicture:null,token:null}}


// si user auth info null li render