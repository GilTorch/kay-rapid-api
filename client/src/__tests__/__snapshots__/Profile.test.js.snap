// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> renders correctly 1`] = `
<div>
  <header
    className="header-bar"
  >
    
    <h1>
      Pwofil Ou
    </h1>
  </header>
  <nav
    className="navigation-bar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="navigation-button"
      >
        <img
          src="home-icon.svg"
        />
        <span
          className="navigation-button__title"
        >
          Akey
        </span>
      </button>
    </a>
    <a
      href="/favorites"
      onClick={[Function]}
    >
      <button
        className="navigation-button"
      >
        <img
          src="favorites-icon.svg"
        />
        <span
          className="navigation-button__title"
        >
          Favori
        </span>
      </button>
    </a>
    <a
      href="/profile"
      onClick={[Function]}
    >
      <button
        className="profile-icon--hightlighted navigation-button--highlighted"
      >
        <img
          src="profile-icon-highlighted.svg"
        />
        <span
          className="navigation-button__title"
        >
          Pwofil
        </span>
      </button>
    </a>
  </nav>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <header
    className="header-bar"
  >
    
    <h1>
      Pwofil Ou
    </h1>
  </header>
  <nav
    className="navigation-bar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="navigation-button"
      >
        <img
          src="home-icon.svg"
        />
        <span
          className="navigation-button__title"
        >
          Akey
        </span>
      </button>
    </a>
    <a
      href="/favorites"
      onClick={[Function]}
    >
      <button
        className="navigation-button"
      >
        <img
          src="favorites-icon.svg"
        />
        <span
          className="navigation-button__title"
        >
          Favori
        </span>
      </button>
    </a>
    <a
      href="/profile"
      onClick={[Function]}
    >
      <button
        className="profile-icon--hightlighted navigation-button--highlighted"
      >
        <img
          src="profile-icon-highlighted.svg"
        />
        <span
          className="navigation-button__title"
        >
          Pwofil
        </span>
      </button>
    </a>
  </nav>
</div>
`;
